### PR TITLE
AppConfiguration 2024-05-01

### PR DESCRIPTION
**Pull request by hahahahahaiyiwen**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [x] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories")
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

**Review by jimmyca15 — APPROVED** [review]

File: specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2024-05-01/appconfiguration.json [hunk 1]
```diff
@@ -0,0 +1,2890 @@
...
+            "$ref": "./examples/ConfigurationStoresDeleteReplica.json"
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "azure-async-operation"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AppConfiguration/configurationStores/{configStoreName}/snapshots/{snapshotName}": {
```
Comment (raosuhas) [inline specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2024-05-01/appconfiguration.json:1384]: do you no support delete for this resource ?

File: specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2024-05-01/appconfiguration.json [hunk 2]
```diff
@@ -0,0 +1,2890 @@
...
+          "type": "integer",
+          "format": "int64",
+          "readOnly": true
+        },
+        "tags": {
+          "description": "The tags of the snapshot. NOTE: These are data plane tags, not ARM tags.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
```
Comment (raosuhas) [inline specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2024-05-01/appconfiguration.json:2793]: please rename to reflect that these are dataplane tags

File: specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2024-05-01/appconfiguration.json [hunk 3]
```diff
@@ -0,0 +1,2890 @@
...
+          "type": "string",
+          "readOnly": true
+        },
+        "properties": {
+          "$ref": "#/definitions/SnapshotProperties",
+          "description": "All snapshot properties.",
+          "x-ms-client-flatten": true
+        }
+      },
+      "x-ms-azure-resource": true
```
Comment (raosuhas) [inline specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2024-05-01/appconfiguration.json:2690]: use proxyresource definition from Commontypes

File: specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2024-05-01/appconfiguration.json (hunk 1 above)
Comment (hahahahahaiyiwen) [reply specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2024-05-01/appconfiguration.json:1384]: this resource is designed to be "PUT" to "Archived" state and then got deleted after certain period of time.

File: specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2024-05-01/appconfiguration.json (hunk 2 above)
Comment (hahahahahaiyiwen) [reply specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2024-05-01/appconfiguration.json:2793]: this "tags" is a well-defined concept already existing in our api spec

File: specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2024-05-01/appconfiguration.json (hunk 3 above)
Comment (hahahahahaiyiwen) [reply specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2024-05-01/appconfiguration.json:2690]: We are stabilizing the preview api, this definition is consistent with what we have in hte existing preview version api spec. [link](https://github.com/Azure/azure-rest-api-specs/blob/2e5be0e72597c6fc8d438f20e38087d900c16427/specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/preview/2023-09-01-preview/appconfiguration.json#L2709C7-L2709C34)

File: specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2024-05-01/appconfiguration.json [hunk 4]
```diff
@@ -0,0 +1,2890 @@
...
+            "name": "AuthenticationMode",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "Local",
+                "description": "The local authentication mode. Users are not required to have data plane permissions if local authentication is not disabled."
+              },
+              {
+                "value": "Pass-through",
+                "description": "The pass-through authentication mode. User identity will be passed through from ARM, requiring user to have data plane action permissions (Available via App Configuration Data Owner/ App Configuration Data Reader)."
```
Comment (razvanbadea-msft) [inline specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2024-05-01/appconfiguration.json:1692]: >ARM [](http://example.com/codeflow?start=112&length=3)

in descriptions you should expand the acronyms

File: specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2024-05-01/appconfiguration.json [hunk 5]
```diff
@@ -0,0 +1,2890 @@
...
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "Snapshot": {
+      "description": "The snapshot resource.",
+      "type": "object",
+      "properties": {
+        "id": {
```
Comment (razvanbadea-msft) [inline specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2024-05-01/appconfiguration.json:2669]: >id [](http://example.com/codeflow?start=9&length=2)

can you use arm-id as format for this property?

Comment (hahahahahaiyiwen) [reply specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2024-05-01/appconfiguration.json:2669]: Updated

File: specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2024-05-01/appconfiguration.json (hunk 4 above)
Comment (hahahahahaiyiwen) [reply specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2024-05-01/appconfiguration.json]: Updated to Azure Resource Manager (ARM)

**Review by jimmyca15 — APPROVED** [review]

**Review by austintolani — APPROVED** [review]

**Pull request merged**